src/models/tools/table/table-content.ts

```diff
@@ -17,16 +17,26 @@ export function defaultTableContent() {
                           } as any);
 }
 
+export interface IColumnProperties {
+  name: string;
+}
+
+export interface IRowProperties {
+  [key: string]: any;
+}
+
 export interface ITableProperties {
-  [prop: string]: any;
+  columns?: IColumnProperties[];
+  rows?: IRowProperties[];
+  beforeId?: string | string[];
+  name?: string;
 }
 
 export interface ITableChange {
   action: "create" | "update" | "delete";
   target: "rows" | "columns";
   ids?: string | string[];
-  props?: ITableProperties | ITableProperties[];
-  others?: ITableProperties;
+  props?: ITableProperties | IColumnProperties | IRowProperties[];
```

Comment:
New format looks much nicer! One main question:

It seems like this lets you create invalid change objects, e.g. 
```
{
action: "create",
target: "columns",
props: {"foo": "bar"}
}
```
I'm wondering if it would be cleaner to have `RowChange`, `ColumnChange` and `CoreTableChange` interfaces instead, with a `TableChange` type that unions them. That could also let you get rid of the `props` field which seems undesirable